parser/yy_parser.go

```diff
@@ -25,7 +30,33 @@ const (
 	CodeSyntaxErr terror.ErrCode = iota + 1
 )
 
-// YYParse is an wrapper of `yyParse` to make it exported.
-func YYParse(yylex yyLexer) int {
-	return yyParse(yylex)
+// Parse parses a query string to raw ast.StmtNode.
+// If charset and collation is "", default charset and collation will be used.
+func Parse(sql, charset, collation string) ([]ast.StmtNode, error) {
+	if charset == "" {
+		charset = mysql.DefaultCharset
+	}
+	if collation == "" {
+		collation = mysql.DefaultCollationName
+	}
+	l := NewLexer(sql)
+	l.SetCharsetInfo(charset, collation)
+	yyParse(l)
```

Comment:
We have `Errors()` method to check error, which is enough, the integer return code is redundant.